landlab/components/soil_grading/soil_grading.py

```diff
@@ -0,0 +1,540 @@
+        """
+
+        super().__init__(grid)
+
+        self._grading_name = grading_name
+        self._meansizes = meansizes
+        self._alpha = alpha
+        self._A_factor = A_factor
+
+        if self._meansizes is not None:
+            self._n_sizes = np.size(self._meansizes)
+            self._grain_max_size = np.max(self._meansizes)
+            self._input_sizes_flag = True
+
+        else:
+            self._input_sizes_flag = False
+            self._n_sizes = n_of_grainsize_classes
+            self._grain_max_size = grain_max_size
+
+        self._N = int(self._grading_name.split("-")[0][1:])
+        self._soil_density = float(soil_density)
+        self._phi = phi
+        self._power_of = power_of
+        self._initial_median_size = initial_median_size
+        self._CV = CV
+        self._initial_total_soil_weight = initial_total_soil_weight
+        if std is None:
+            std = self._initial_median_size * self._CV
+        self._std = std
+        self._is_bedrock_distribution_flag = is_bedrock_distribution_flag
+        self._precent_of_volume_in_spread = precent_of_volume_in_spread
+
+        # Note: Landlabs' init_out_field procedure will not work
+        # for the 'grains__weight' and 'grains_classes__size' fields
+        # because the shape of these fields is: n_nodes x n_grain_sizes.
+        grid.add_field(
+            "median_size__weight",
+            np.zeros(grid.shape[0] * grid.shape[1]),
+            at="node",
+            dtype=float,
+        )
+        grid.add_field(
+            "grains_classes__size",
+            np.ones((int(grid.shape[0] * grid.shape[1]), self._n_sizes)),
+            at="node",
+            dtype=float,
+        )
+        grid.add_field(
+            "grains__weight",
+            np.zeros((int(grid.shape[0] * grid.shape[1]), self._n_sizes)),
+            at="node",
+            dtype=float,
+        )
+        grid.add_field(
+            "bed_grains__proportions",
+            np.ones((int(grid.shape[0] * grid.shape[1]), self._n_sizes)),
+            at="node",
+            dtype=float,
+        )
+
+        try:
+            grid.add_field(
+                "soil__depth",
+                np.zeros(int(grid.shape[0] * grid.shape[1])),
+                at="node",
+                dtype=float,
+            )
+        except KeyError as exc:
+            raise ValueError("Soil field already exists") from exc
+
+        if "topographic__elevation" not in grid.at_node:
+            grid.add_field(
+                "topographic__elevation",
+                np.zeros_like(self._grid.nodes.flatten(), dtype=float),
+                at="node",
+                dtype=float,
+            )
+
+        if "bedrock__elevation" not in grid.at_node:
+            grid.add_field(
+                "bedrock__elevation",
+                np.zeros_like(self._grid.nodes.flatten(), dtype=float),
+                at="node",
+                dtype=float,
+            )
```

Comment:
As above, use `grid.add_zeros`. It's fewer lines of code and avoids creating an temporary array.

Reply:
Changed